src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1586,7 +1619,7 @@ private async Task WriteComplexAndExpandedNavigationPropertyAsync(IEdmProperty e
 
             object propertyValue = resourceContext.GetPropertyValue(edmProperty.Name);
 
-            if (propertyValue == null || propertyValue is NullEdmComplexObject)
+            if (propertyValue == null || propertyValue is NullEdmComplexObject || propertyValue is ODataIdContainer)
```

Comment:
Curious if `is` would throw an exception the l-value is null... If no, the null check is superfluous

Reply:
`Logical OR` is evaluated from left to right. If the `propertyValue == null` is `true`, the rest are not evaluated.
See https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-or-operator-

Reply:
I don't think you understood what I meant...
Let me rephrase
Would the following throw an exception if `propertyValue` is null?
```csharp
if (propertyValue is NullEdmComplexObject)
{
    // ...
}
```

Reply:
I think from his explanation, he means if `propertyValue==null` then

```csharp
if (propertyValue is NullEdmComplexObject)
{
    // ...
}
```
is not evaluated